uber/models.py

```diff
@@ -593,27 +593,12 @@ def auto_badge_num(self, badge_type):
 
                 # Searches badge range for a gap in badge numbers; if none found, returns the latest badge number + 1
                 # Doing this lets admins manually set high badge numbers without filling up the badge type's range.
-                first = 0
-                last = sametype.count() - 1
-                if sametype_list[last] - sametype_list[first] == last:
+                start, end = c.BADGE_RANGES[badge_type][0], sametype_list[-1]
+                gap_nums = sorted(set(range(start, end + 1)).difference(sametype_list))
```

Comment:
This is the magic that replaces the entire while loop below.

It uses a very handy (and previously unknown to me) Python function called `difference`, which takes set A and set B and tells you what is in set A that is not in set B. We create a set of numbers that go from the badge range start (not the start of the list, because it could theoretically be higher than the first available badge number) to the end of list of numbers, and compare that with the actual list of badge numbers.

My only question is performance, but everything I can find on Google about sets indicates that they are well-optimized. I Am Not An Expert but people keep praising their look-up abilities, especially when compared with lists. I don't know if that means they're faster than a binary search, but hopefully they're Fast Enough.